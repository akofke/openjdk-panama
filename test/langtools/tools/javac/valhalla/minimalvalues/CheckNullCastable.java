/*
 * @test /nodynamiccopyright/
 * @summary null cannot be casted to and compared with value types.
 * @modules jdk.incubator.mvt
 * @compile/fail/ref=CheckNullCastable.out --should-stop:ifError=PARSE -XDrawDiagnostics -Werror  -Xlint:values  CheckNullCastable.java
 */
@jdk.incubator.mvt.ValueCapableClass
final class CheckNullCastable {
    void foo(CheckNullCastable cnc) {
        CheckNullCastable cncl = (CheckNullCastable) null;
        if (cnc != null) {};
        if (null != cnc) {};
    }
}
