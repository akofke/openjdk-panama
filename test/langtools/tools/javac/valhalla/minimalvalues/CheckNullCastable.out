- compiler.warn.incubating.modules: jdk.incubator.mvt
- compiler.err.warnings.and.werror
CheckNullCastable.java:10:54: compiler.warn.prob.found.req: (compiler.misc.inconvertible.types: compiler.misc.type.null, CheckNullCastable)
CheckNullCastable.java:11:17: compiler.warn.incomparable.types: CheckNullCastable, compiler.misc.type.null
CheckNullCastable.java:11:17: compiler.warn.value.does.not.support: !=
CheckNullCastable.java:12:18: compiler.warn.incomparable.types: compiler.misc.type.null, CheckNullCastable
CheckNullCastable.java:12:18: compiler.warn.value.does.not.support: !=
1 error
6 warnings
